00-framework-02-document-header.md

```diff
@@ -0,0 +1,157 @@
+# Document Header for libp2p Specs
+
+> A standard document header to indicate spec maturity, status & ownership
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+|-----------------|---------------|--------|-----------------|
+| 1A              | Working Draft | Active | r0, 2019-05-27  |
+
+Authors: [@yusefnapora]
+
+Interest Group: TBD
```

Comment:
Maybe declare 5, since this spec looks pretty finalised and promotion to Candidate Recommendation and then Recommendation should be imminent? Suggestions: @raulk, @vyzo, @mgoelzer, @jacobheun, @tomaka?